pymc_marketing/mmm/fourier.py

```diff
@@ -261,63 +259,52 @@ def generate_fourier_modes(
     )
 
 
-class FourierBase:
+class FourierBase(BaseModel):
     """Base class for Fourier seasonality transformations.
 
     Parameters
     ----------
     n_order : int
         Number of fourier modes to use.
+    days_in_period : float
+        Number of days in a period.
     prefix : str, optional
         Alternative prefix for the fourier seasonality, by default None or
         "fourier"
     prior : Prior, optional
         Prior distribution for the fourier seasonality beta parameters, by
-        default None
-    name : str, optional
-        Name of the variable that multiplies the fourier modes, by default None
-
-    Attributes
-    ----------
-    days_in_period : float
-        Number of days in a period.
-    prefix : str
-        Name of model coordinates
-    default_prior : Prior
-        Default prior distribution for the fourier seasonality
-        beta parameters.
+        default `Prior("Laplace", mu=0, b=1)`
+    variable_name : str, optional
+        Name of the variable that multiplies the fourier modes. By default None,
+        in which case it is set to the `{prefix}_beta`.
 
     """
 
-    days_in_period: float
-    prefix: str = "fourier"
+    n_order: int = Field(..., gt=0)
+    days_in_period: float = Field(..., gt=0)
+    prefix: str = Field("fourier")
+    prior: InstanceOf[Prior] = Field(Prior("Laplace", mu=0, b=1))
+    variable_name: str | None = Field(None)
 
-    default_prior = Prior("Laplace", mu=0, b=1)
-
-    def __init__(
-        self,
-        n_order: int,
-        prefix: str | None = None,
-        prior: Prior | None = None,
-        name: str | None = None,
-    ) -> None:
-        if not isinstance(n_order, int) or n_order < 1:
-            raise ValueError(f"n_order must be a positive integer. Not {n_order}")
-
-        self.n_order = n_order
-        self.prefix = prefix or self.prefix
-        self.prior = prior or self.default_prior
-        self.variable_name = name or f"{self.prefix}_beta"
-
-        if self.variable_name == self.prefix:
-            raise ValueError("Variable name cannot be the same as the prefix")
+    def model_post_init(self, __context: Any) -> None:
+        if self.variable_name is None:
+            self.variable_name = f"{self.prefix}_beta"
 
         if not self.prior.dims:
             self.prior = self.prior.deepcopy()
             self.prior.dims = self.prefix
 
+    @model_validator(mode="after")
+    def check_variable_name(self) -> Self:
+        if self.variable_name == self.prefix:
+            raise ValueError("Variable name cannot be the same as the prefix")
+        return self
+
+    @model_validator(mode="after")
+    def check_prior_has_right_dimensions(self) -> Self:
```

Comment:
Same public method comment for this

Reply:
also in https://github.com/pymc-labs/pymc-marketing/pull/809/commits/32d9118779577680a77b462e4984c8da62274e61